Universal/Sniffs/DeclareStatements/DeclareStatementsStyleSniff.php

```diff
@@ -0,0 +1,241 @@
+ * and require or allow the style to only one of the two directives (`ticks` and `encoding`).
+ *
+ * @since 1.0.0
+ */
+class DeclareStatementsStyleSniff implements Sniff
+{
+
+    /**
+     * Whether to require curly brace style or disallow it
+     * when using declare statements.
+     *
+     * Possible values: requireBraces, disallowBraces.
+     *
+     * Default is disallowBraces.
+     *
+     * @var string
+     */
+    public $declareStyle = 'disallowBraces';
+
+    /**
+     * Which declare to enforce the brace style for.
+     * If not explicitly defined, the brace style will be applied
+     * to all declare directive.
+     *
+     * The exception is the strict_types directive which cannot be
+     * written using curly braces.
+     *
+     * Possible values: ticks, encoding.
+     *
+     * @var array
+     */
+    public $directiveType = [];
+
+    /**
+     * Allowed declare directives.
+     *
+     * @var array
+     */
+    private $allowedDirectives = [
+        'strict_types',
+        'ticks',
+        'encoding',
+    ];
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @since 1.0.0
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [
+            T_DECLARE
+        ];
+    }
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @since 1.0.0
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        $openParenPtr  = $tokens[$stackPtr]['parenthesis_opener'];
+        $closeParenPtr = $tokens[$stackPtr]['parenthesis_closer'];
+
+        if (isset($openParenPtr, $closeParenPtr) === false) {
+            // Parse error or live coding, bow out.
+            return;
+        }
+
+        $directiveStrings = [];
+        // Get the next string and check if it's an allowed directive.
+        // Find all the directive strings inside the declare statement.
+        for ($i = $openParenPtr; $i <= $closeParenPtr; $i++) {
+            if ($tokens[$i]['code'] === \T_STRING) {
+                $directiveStrings[] = $tokens[$i]['content'];
+            }
+        }
+
+        foreach ($directiveStrings as $directiveString) {
+            if (!in_array($directiveString, $this->allowedDirectives, true)) {
+                $phpcsFile->addError(
+                    sprintf(
+                        'Declare directives can be one of: %1$s. "%2$s" found.',
+                        implode(', ', $this->allowedDirectives),
+                        $directiveString
+                    ),
+                    $stackPtr,
+                    'WrongDeclareDirective'
```

Comment:
Again, not really relevant anymore, but just something to think over:

```suggestion
                    'InvalidDirective'
```